src/main/java/bio/terra/service/profile/flight/create/GetOrCreateProfileIdStep.java

```diff
@@ -0,0 +1,29 @@
+package bio.terra.service.profile.flight.create;
+
+import bio.terra.model.BillingProfileRequestModel;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import java.util.UUID;
+
+public class GetOrCreateProfileIdStep implements Step {
+  private final BillingProfileRequestModel request;
+
+  public GetOrCreateProfileIdStep(BillingProfileRequestModel request) {
+    this.request = request;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException {
+    UUID profileId = request.getId();
+    if (profileId == null) {
```

Comment:
Should we do an empty string check since a user could also pass that in?  I think `StringUtils.isEmpty` handles both null and empty string...IMO both mean "not specified by the user"

Reply:
It looks like we might catch an empty id w/ the null check. 
I tried created a profile with "id": "" and ran into the following errors:
```
"message": "Validation errors - see error details",
  "errorDetail": [
    "id: 'NotNull' (must not be null)",
    "id: 'The billing profile id must be specified'"
  ]
```
The last error message lines up with the validation error above in the ProfileRequestValidator.

Reply:
Yeah! And now this will will generate an id in that case instead of throwing an error.

Reply:
oh great, ok nevermind then!